controller/model/service_handlers.go

```diff
@@ -121,76 +119,64 @@ func (handler *ServiceHandler) ReadForIdentityInTx(tx *bbolt.Tx, id string, iden
+func (handler *EdgeServiceHandler) PublicQueryForIdentity(sessionIdentity *Identity, configTypes map[string]struct{}, query ast.Query) (*ServiceListResult, error) {
 	if sessionIdentity.IsAdmin {
-		return handler.listServices(queryOptions, sessionIdentity.Id, configTypes, true)
+		return handler.queryServices(query, sessionIdentity.Id, configTypes, true)
 	}
-	query := queryOptions.Predicate
-	if query != "" {
-		query = "(" + query + ") and "
-	}
-	query += fmt.Sprintf(`not isEmpty(from servicePolicies where anyOf(identities) = "%v")`, sessionIdentity.Id)
-	queryOptions.finalQuery = query
-	return handler.listServices(queryOptions, sessionIdentity.Id, configTypes, false)
-}
-
-func (handler *ServiceHandler) queryServices(tx *bbolt.Tx, query string) (*ServiceListResult, error) {
-	result := &ServiceListResult{handler: handler}
-	err := handler.listWithTx(tx, query, result.collect)
+	idFilterQueryString := fmt.Sprintf(`not isEmpty(from servicePolicies where anyOf(identities) = "%v")`, sessionIdentity.Id)
+	idFilterQuery, err := ast.Parse(handler.Store, idFilterQueryString)
 	if err != nil {
 		return nil, err
 	}
-	return result, nil
+
+	query.SetPredicate(ast.NewAndExprNode(query.GetPredicate(), idFilterQuery.GetPredicate()))
+
+	return handler.queryServices(query, sessionIdentity.Id, configTypes, false)
 }
 
-func (handler *ServiceHandler) listServices(queryOptions *QueryOptions, identityId string, configTypes map[string]struct{}, isAdmin bool) (*ServiceListResult, error) {
+func (handler *EdgeServiceHandler) queryServices(query ast.Query, identityId string, configTypes map[string]struct{}, isAdmin bool) (*ServiceListResult, error) {
 	result := &ServiceListResult{
 		handler:     handler,
 		identityId:  identityId,
 		configTypes: configTypes,
 		isAdmin:     isAdmin,
 	}
-	err := handler.parseAndList(queryOptions, result.collect)
+	err := handler.preparedList(query, result.collect)
 	if err != nil {
 		return nil, err
 	}
 	return result, nil
 }
 
-func (handler *ServiceHandler) CollectServiceEdgeRouterPolicies(id string, collector func(entity BaseModelEntity)) error {
-	return handler.collectAssociated(id, persistence.EntityTypeServiceEdgeRouterPolicies, handler.env.GetHandlers().ServiceEdgeRouterPolicy, collector)
-}
-
-func (handler *ServiceHandler) CollectServicePolicies(id string, collector func(entity BaseModelEntity)) error {
-	return handler.collectAssociated(id, persistence.EntityTypeServicePolicies, handler.env.GetHandlers().ServicePolicy, collector)
-}
-
-func (handler *ServiceHandler) CollectConfigs(id string, collector func(entity BaseModelEntity)) error {
-	return handler.collectAssociated(id, persistence.EntityTypeConfigs, handler.env.GetHandlers().Config, collector)
+func (handler *EdgeServiceHandler) GetRoleAttributes() []string {
+	//handler.GetDb().View(func(tx *bbolt.Tx) error {
+	//	return nil
+	//})
```

Comment:
Was this intentional?

Reply:
I was working on plumbing the role attributes list functions and this got left behind. I'll remove it